src/bindings/python/flux/future.py

```diff
@@ -115,6 +115,9 @@ def then(self, callback, arg=None, timeout=-1.0):
         # For example `f.rpc('topic').then(cb).wait_for(-1)
         return self
 
+    def destroy(self):
+        self.pimpl.destroy()
+
```

Comment:
I'd prefer to not include this public method.  I previously attempted to make this work, but exposing it publically opens up all sorts of edge-cases.  I'd prefer to leverage the `_clear` method in the `Wrapper` class.